tensorflow/python/ops/variable_scope.py

```diff
@@ -662,8 +662,13 @@ def _get_single_variable(self,
         init_val = initializer
         variable_dtype = None
       else:
-        init_val = lambda: initializer(
-            shape.as_list(), dtype=dtype, partition_info=partition_info)
+        if partition_info is not None:
```

Comment:
Hey, thanks a bunch your comments! Not entirely sure how to fix those things you mentioned though. Maybe I'm better off avoiding hacks to add backward compatibility if it ends up messing things up.